dart/lib/src/sentry_options.dart

```diff
@@ -160,8 +160,18 @@ class SentryOptions {
   /// Whether to send personal identifiable information along with events
   bool sendDefaultPii = false;
 
-  SentryOptions({this.dsn}) {
+  SentryOptions({this.dsn, PlatformChecker? checker}) {
     sdk.addPackage('pub:sentry', sdkVersion);
+    if (checker != null) {
+      platformChecker = checker;
+    }
+
+    // In debug mode we want to log everything by default to the console.
+    // In order to do that, this must be the first thing the SDK does
+    // and the first thing the SDK does, is to instantiate SentryOptions
+    if (platformChecker.isDebugMode() && logger == noOpLogger) {
+      _logger = dartLogger;
+    }
```

Comment:
so I think this works automatically, but what's about if the user changed the `debug` flag manually in the options callback, we still need to take that into consideration.

eg: user mutate the debug flag to `true` or `false`, and it does not matter the Flutter `debugMode`

Reply:
That does nothing to the logger, as it happens after the logger was set.
What do you expect to happen if the user sets debug to true/false?

Reply:
the final decision of the user always has precedence over automatic configurations.
right now if App is in debug mode, it logs everything to the console.
if user init. the Sdk and set `debug` to false, it still logs everything (I guess)? if it works as intended, then please ignore my comment, I might have been mistaken too.

Reply:
Yep, currently if the user sets debug to false it still logs everything.

Sooo we should then set the logger again after the sentry init callback. That way no logs get dropped between Sentry initialization and setting the logger based on the users settings. Does that make sense?

Reply:
yeah that would be the behaviour I'd expect